src/main/java/codeu/controller/LogoutServlet.java

```diff
@@ -0,0 +1,40 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package codeu.controller;
+
+import java.io.IOException;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet class responsible for the logout page. */
+public class LogoutServlet extends HttpServlet {
+
+  /**
+   * This function fires when a user requests the /login URL. It simply forwards the request to
+   * login.jsp.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+      throws IOException, ServletException {
+        
+    if(request.getSession(false) != null){
```

Comment:
Checking for a null session may not be sufficient. I believe it is possible to have a non-null session, but not be "logged in" (i.e. have the user attribute set). Try requesting the logout page again after logging out to see what I mean.